Final Task

vitaliesirbu opened the pull request:
Implementing Account registration pageobject and test

vitaliesirbu committed:
Final Task
Implementing Account registration pageobject and test

vitaliesirbu committed:
some minro changes.

vitaliesirbu committed:
Cleaning the project for old classes
Implementing new user credentials class
Implementing the use of Util class

vitaliesirbu committed:
Implementing User class and UserFactory pattern

vitaliesirbu committed:
IMplementing methods for each field in AccountCreationPage
Add the LOCATOR_TIMEOUT constant

vitaliesirbu committed:
createAcount method returns MyAccountPage instance

vitaliesirbu committed:
Adding a Wait functionality to isElementDisplayed method

vitaliesirbu committed:
Refactoring ElementUtils by moving Wait directly in return statement

vitaliesirbu committed:
Implementing LoginTest

shukal94 commented on src/main/java/com/coherentsolutions/training/automation/web/sirbu/pageobjects/AccountCreationPage.java:
```diff
@@ -0,0 +1,87 @@
+
+    }
+
+    public void fillInFirstNameField(User user){
+        firstNameField.sendKeys(user.getFirstName());
+    }
+    public void fillInLastNameField(User user){
+        lastNameField.sendKeys(user.getLastName());
+    }
+    public void fillInEmailField(User user){
+        emailField.sendKeys(user.getEmail());
+    }
+    public void fillInPasswordField(User user){
+        passwordField.sendKeys(user.getPassword());
+    }
+    public void fillInConfirmPasswordField(User user){
+        confirmPasswordField.sendKeys(user.getPassword());
+    }
+    public void clickOnConfirmButton(){
+        createAccountButton.click();
+    }
```
let's not use void here, and therefore for PO methods

shukal94 commented on src/main/java/com/coherentsolutions/training/automation/web/sirbu/pageobjects/AccountCreationPage.java:
```diff
@@ -0,0 +1,87 @@
+        createAccountButton.click();
+    }
+
+    public String actualAccountName(WebElement actual){
```
just getAccountName

shukal94 commented on src/main/java/com/coherentsolutions/training/automation/web/sirbu/pageobjects/HomePage.java:
```diff
@@ -1,52 +1,48 @@
 
+    public HomePage(WebDriver driver){
+        super(driver);
+        PageFactory.initElements(new AjaxElementLocatorFactory(driver, LOCATOR_TIMEOUT), this);
```
let's try to move this call to super constructor

vitaliesirbu committed:
avoiding the use of Void in methods
moving PageFactory.initElements super contructor
renaming actualAccountName to getAccountName

vitaliesirbu committed:
removing all remaining Void in PO methods

vitaliesirbu committed:
Adding Address Page, Create New Address Page, and Test
Updating User class to incorporate Adress class.

vitaliesirbu committed:
Minor changes after debug

shukal94 commented on src/test/java/com/coherentsolutions/training/automation/web/sirbu/AccountCreationTest.java:
```diff
@@ -0,0 +1,56 @@
+
+        accountCreationPage.createAccount(baseUser);
+
+        WebElement contactInformation = driver.findElement(By.cssSelector(".box-content p"));
```
don't interact with we element in tests. use pages instead

shukal94 commented on src/test/java/com/coherentsolutions/training/automation/web/sirbu/AddNewAddressTest.java:
```diff
@@ -0,0 +1,52 @@
+    @Epic("User Settings")
+    @Feature("Adress")
+    @Story("Creating new address")
+    public void testAddNewAddress(){
```
add some assertions here

vitaliesirbu committed:
Adding Address Assertions

vitaliesirbu committed:
Code Formating
